source/features/stake-pools/components/StakePoolTooltip.tsx

```diff
@@ -0,0 +1,138 @@
+import classnames from 'classnames';
+import moment from 'moment';
+import React, { FC, useCallback, useEffect, useRef } from 'react';
+import { MouseEvent } from 'react';
+import { getColorFromRange } from '../../../utils/colors';
+import { IStakePoolTooltipProps } from '../types';
+import styles from './StakePoolTooltip.scss';
+
+const CloseCrossIcon = require('../../../static/assets/images/stake-pools/close-cross.svg');
+const ExternalLinkIcon = require('../../../static/assets/images/stake-pools/link-ic.svg');
+
+const StakePoolTooltip: FC<IStakePoolTooltipProps> = ({
+  stakePool,
+  onClose,
+  position,
+  color,
+}) => {
+  const {
+    name,
+    description,
+    ticker,
+    url,
+    ranking,
+    controlledStake,
+    profitMargin,
+    performance,
+    retiring,
+  } = stakePool;
+
+  const darken = 1;
+  const alpha = 0.3;
+  const reverse = true;
+  const retirementFromNow = retiring ? moment(retiring).fromNow(true) : '';
+  const colorBand = getColorFromRange(ranking);
+
+  const colorBandStyle = {
+    backgroundColor: colorBand,
+  };
+
+  const stakePoolTooltipStyles = classnames([
+    styles.stakePoolTooltipContainer,
+    styles[position.vertical],
+    styles[position.horizontal],
+  ]);
+
+  const handleOuterClick = useCallback((event: Event) => {
+    const target = event.target as HTMLElement;
+    if (tooltipNode.current && !tooltipNode.current.contains(target)) {
+      onClose();
+    }
+  }, []);
+
+  useEffect(() => {
```

Comment:
@daniloprates Not critical, but it would be better if you extract this part to custom Hook.

Reply:
I'll open sub-cards for any adjustment. Thanks!